public/app/plugins/panel/graph3/configFactory.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  DataFrame,
+  FieldConfigSource,
+  FieldMatcherID,
+  getFieldDisplayName,
+  isSystemOverride,
+  SystemConfigOverrideRule,
+} from '@grafana/data';
+import { GraphNGLegendEvent, GraphNGLegendEventMode } from '@grafana/ui';
+
+const displayOverrideRef = 'series_display';
+const isDisplayOverride = isSystemOverride(displayOverrideRef);
+
+export const displayConfigFactory = (
+  event: GraphNGLegendEvent,
+  fieldConfig: FieldConfigSource<any>,
+  data: DataFrame[]
+): FieldConfigSource<any> => {
+  const { fieldIndex, mode } = event;
+  const { overrides } = fieldConfig;
+
+  const frame = data[fieldIndex.frameIndex];
+  const field = frame.fields[fieldIndex.fieldIndex];
+  const displayName = getFieldDisplayName(field, frame, data);
+  const currentIndex = overrides.findIndex(isDisplayOverride);
+
+  if (currentIndex < 0) {
+    const override = createFreshOverride(displayName);
+
+    return {
+      ...fieldConfig,
+      overrides: [override, ...fieldConfig.overrides],
+    };
+  }
+
+  const overridesCopy = Array.from(overrides);
+  const [current] = overridesCopy.splice(currentIndex, 1) as SystemConfigOverrideRule[];
+
+  if (mode === GraphNGLegendEventMode.select) {
+    const existing = matchersInConfig(current);
+
+    if (existing.find(name => name === displayName)) {
+      return {
+        ...fieldConfig,
+        overrides: overridesCopy,
+      };
+    }
+
+    const override = createFreshOverride(displayName);
+
+    return {
+      ...fieldConfig,
+      overrides: [override, ...overridesCopy],
+    };
+  }
+
+  const override = createExtendedOverride(current, displayName);
+  console.log('override', override);
+
+  return {
+    ...fieldConfig,
+    overrides: [override, ...overridesCopy],
+  };
+};
+
+const createFreshOverride = (displayName: string): SystemConfigOverrideRule => {
+  return {
+    __systemRef: displayOverrideRef,
+    matcher: {
+      id: FieldMatcherID.readOnly,
+      options: {
+        innerId: FieldMatcherID.byRegexp,
+        innerOptions: `^(?!${displayName}$).*$`,
+        formattedValue: `All series except: ${displayName}`,
+      },
```

Comment:
I really like the idea of readonly matcher that actually takes other matcher as an option, very clever.

1. What I'm not sure about tho is whether or not we should have the formattedValue being displayed or rather the matcher editor with read-only property. Comparing the UI of the two alternatives I find the editor working better with the read-only input making the entire UI more consistent. 
![image](https://user-images.githubusercontent.com/2376619/102220130-b11d0e80-3ee0-11eb-8b2d-1d027ce6ef26.png)
vs
![image](https://user-images.githubusercontent.com/2376619/102220218-cdb94680-3ee0-11eb-8964-a1949ba66804.png)
However, the big advantage of your proposal that the matcher rule is human readable (I'm not saying that only robots can read regex ;) )

2. Another thing regarding the _system_ overrides is whether or not we should allow multiple properties to be overridden for such an override. I mean, I think the `Add override property`  and property removal buttons should be disabled IMO:
<img width="595" alt="Screen Shot 2020-12-15 at 14 24 11" src="https://user-images.githubusercontent.com/2376619/102220625-6059e580-3ee1-11eb-96b5-bf0e7fd3a9a3.png">

3. For the read-only / system override I think we should add some kind of indication or maybe a tooltip why people can not edit those. Maybe @jessover9000 or @hemdrup could help us with this? 

In general I feel like the ui of the system/readonly overrides is something worth bringing to the UX feedback session tomorrow.